### DISCUSSION TITLE
Getting unexpected validation error with optional field

### DISCUSSION THREAD

**Discussion by balukrishnans** [pydantic/pydantic#4050] [Question]
``` py
from typing import Optional, List
from pydantic import BaseModel, Field

class Foo(BaseModel):
    bar: Optional[List[int]] = Field(None, unique_items=True)

obj = Foo(bar=None)
```
I am getting errors like
```
Traceback (most recent call last):
  File "test.py", line 8, in <module>
    obj = Foo(bar=None)
  File "pydantic/main.py", line 331, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Foo
bar
  'NoneType' object is not iterable (type=type_error)
```

My expectation is
- `bar` is an optional field and can have a None value.
- specified `unique_items=True` and that validation must only raise an error if the value of the `bar` field is a list type and it's having the same items in it.

Is there any issue with this?

**Comment by LordFckHelmchen** [comment]
Work around would be to use `default_factory=list` instead of `None` with the additional benefit of not having "None poisoning", e.g. checking for None in each place bar is used.

**Comment by balukrishnans** [comment]
Raised one issue related to this discussion. Issue link: https://github.com/samuelcolvin/pydantic/issues/4119
